### PR TITLE
Upgrade template

### LINKED ISSUE #276: feat: Python 3.12 support
Hello team,

This part of the code is failing on Python 3.12:

https://github.com/ethereum/eth-tester/blob/3c76d34beb3ade64ec4569462b36351bdb9a0821/eth_tester/backends/pyevm/utils.py#L5

Do you plan to work on Python 3.12 support (maybe are you already working on it? 🤞🏻 😈)?

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Merge template updates, notably adding python 3.12 support
Closes #276 

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-tester/blob/main/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

**Review by reedsa — APPROVED** [review]
lgtm

File: eth_tester/py.typed
Comment (kclowes) [inline eth_tester/py.typed:1]: is eth_tester typed?

**Review by kclowes — COMMENTED** [review]
I know this is already merged, but I'm not sure we want to add a typed.py without having types? I only looked quickly, so it may have types and I just missed them.

File: eth_tester/py.typed
Comment (pacrob) [reply eth_tester/py.typed:1]: sure ain't

**Comment by wakamex** [comment]
can we get a version bump? this is the only dependency of apeworx that still uses setuptools. pinning to a new version would allow us to remove our own setuptools dependency :)

being tracked here https://github.com/ApeWorX/ape/pull/2003

**Comment by pacrob** [comment]
> can we get a version bump?

Has been released as `v0.11.0-beta.2`

**Pull request merged**